src/timer.rs

```diff
@@ -457,10 +461,7 @@ impl<'a> ParallelSleepDriver<'a> {
             if !is_over(next_timer, now as u32) {
                 break;
             } else {
-                match stop_alarm_at(next_timer.instant as usize) {
-                    Ok(_) => (),
-                    Err(_) => (),
-                }
+                stop_alarm_at(next_timer.instant as usize)?;
```

Comment:
Is this change intentional? The previous code did nothing upon error, whereas the new code returns in that case.

Reply:
No, this is a mistake coming from fixing a lint and I will have to fix this. Good spot!

Reply:
Ah, I tricked myself. It's actually correct, because `EALREADY` is already ignored.